fuel/src/test/kotlin/com/github/kittinunf/fuel/MockHttpTestCase.kt

```diff
@@ -0,0 +1,165 @@
+package com.github.kittinunf.fuel
+
+import org.junit.After
+import org.junit.Before
+import org.mockserver.integration.ClientAndServer
+import org.mockserver.integration.ClientAndServer.startClientAndServer
+import org.mockserver.matchers.Times
+import org.mockserver.model.HttpRequest
+import org.mockserver.model.HttpRequest.request
+import org.mockserver.model.HttpResponse
+import org.mockserver.model.HttpResponse.response
+import org.mockserver.model.HttpTemplate
+import org.mockserver.model.HttpTemplate.template
+import kotlin.text.Typography.times
+
+abstract class MockHttpTestCase : BaseTestCase() {
+
+    private var mockServer: ClientAndServer? = null
+
+    @Before
+    fun setup(){
+        // This is not placed in a @BeforeClass / @BeforeAll so that the tests may have parallel
+        // execution. When there is no port given as first argument, it will grab a free port
+        this.mockServer = startClientAndServer()
```

Comment:
I don't know if we need it; I reckon having a mock server instance per test allows us to isolate test issues, instead of registering requests to one big instance (and the need to reset after each test), or am I missing something?